packages/forklift-console-plugin/src/modules/ProvidersNG/utils/validators/secret/openstackSecretValidator.ts

```diff
@@ -0,0 +1,86 @@
+import { V1Secret } from '@kubev2v/types';
+
+import { missingKeysInSecretData, safeBase64Decode } from '../../helpers';
+import { openstackSecretFieldValidator } from '../secret-fields';
+
+export function openstackSecretValidator(secret: V1Secret) {
+  const authType = safeBase64Decode(secret?.data?.['authType']) || 'password';
+
+  let requiredFields = [];
+  let validateFields = [];
+
+  // guess authenticationType based on authType and username
+  switch (authType) {
+    case 'password':
+      requiredFields = ['username', 'password', 'regionName', 'projectName', 'domainName'];
+      validateFields = [
+        'username',
+        'password',
+        'regionName',
+        'projectName',
+        'domainName',
+        'cacert',
+        'insecureSkipVerify',
+      ];
+      break;
+    case 'token':
+      if (secret?.data?.['username']) {
```

Comment:
@mmartinv hi,
the `authType` is not enough to know what set of fields are needed,
does using `username` makes sense here ?

Reply:
thanks,

> Not sure if it makes sense to support all of them in the UI though...

hmm, we can remove the unneeded ones later

> For authType=token there are two possibilities: provide token, userID, projectID or provide token, username, projectName and userDomainName

the quiestion is, " Can the UI differentiate between the two options by looking at `username` " ?

the current logic is:
a. if authType == 'token'
b. then if username is not empty
        it's token option I
     else
        it's token option II
        
does that logic looks ok for you ?

Reply:
Sure, the logic looks good.

Reply:
thanks :+1: